tortoise/state.go

```diff
@@ -204,29 +199,39 @@ func (v *votes) cutBefore(lid types.LayerID) {
 func (v *votes) find(lid types.LayerID, bid types.BlockID) sign {
 	for current := v.tail; current != nil; current = current.prev {
 		if current.lid == lid {
-			for _, block := range current.blocks {
+			for _, block := range current.supported {
 				if block.id == bid {
-					return block.vote
+					return support
 				}
 			}
+			return against
 		}
 	}
 	return abstain
 }
 
 type layerVote struct {
 	*layerInfo
-	vote   sign
-	blocks []blockVote
+	vote      sign
+	supported []*blockInfo
```

Comment:
why not a map here? does order matter?

Reply:
maps have high overhead. i am trying to avoid using them